src/make_embeddings.jl

```diff
@@ -0,0 +1,165 @@
+
+    # Process each pair of files
+    for (hostname, chunk_files) in hostname_files
+        for (chunk_size, files) in chunk_files
+            if haskey(files, "chunks") && haskey(files, "sources")
+                chunks_file = files["chunks"]
+                sources_file = files["sources"]
+                chunks = deserialize(chunks_file)
+                sources = deserialize(sources_file)
+                cost_tracker = Threads.Atomic{Float64}(0.0)
+                full_embeddings = RT.get_embeddings(embedder, chunks; model="text-embedding-3-large", verbose=false, cost_tracker, api_key=ENV["OPENAI_API_KEY"])
+
+                # Float32
+                fn_output = joinpath(dir_path, "packs", "$hostname-textembedding3large-0-Float32__v1.0.tar.gz")
+                fn_temp = joinpath(dir_path, "packs", "pack.hdf5")
+                h5open(fn_temp, "w") do file
+                    file["chunks"] = chunks
+                    file["sources"] = sources
+                    file["embeddings"] = full_embeddings
+                    file["type"] = "ChunkIndex"
+                    # file["metadata"] = "$hostname ecosystem docstrings, chunk size $chunk_size, downloaded on 20240330, contains: Makie.jl, AlgebraOfGraphics.jl, GeoMakie.jl, GraphMakie.jl, MakieThemes.jl, TopoPlots.jl, Tyler.jl"
+                end
+                run(`tar -cvzf $fn_output -C $(dirname(fn_temp)) $(basename(fn_temp))`)
+                report_artifact(fn_output)
+
+            else
+                @warn "Missing pair for hostname: $hostname, chunk size: $chunk_size"
+            end
+        end
+    end
+
+end
+
+
+
+"""
+    create_output_folders()
+
+Creates output folders
+"""
+function create_output_folders()
+    # Define the folder path    
+    folder_path = joinpath("RAGKit", "knowledge_packs", "packs")
+
+    # Check if the folder exists
+    if !isdir(folder_path)
+        mkpath(folder_path)
+        @info "Folder created: $folder_path"
+    else
+        @info "Folder already exists: $folder_path"
+    end
+
+end
+
+"""
+    make_chunks(hostname_url_dict::Dict{AbstractString,Vector{AbstractString}})
+
+Parses URLs from `hostname_url_dict` and saves the chunks
+
+# Arguments
+- `hostname_url_dict`: Dict with key being hostname and value being a vector of URLs
+"""
+function make_chunks(hostname_url_dict::Dict{AbstractString,Vector{AbstractString}})
+    output_chunks = Vector{SubString{String}}()
+    output_sources = Vector{String}()
+    SAVE_CHUNKS = true
+    CHUNK_SIZE = 512
+    for (hostname, urls) in hostname_url_dict
+        for url in urls
+            try
+                chunks, sources = process_paths(url)
+                append!(output_chunks, chunks)
+                append!(output_sources, sources)
+            catch
+                @error "error!! check url: $url"
+            end
+        end
+        if SAVE_CHUNKS
+            serialize(joinpath("RAGKit", "knowledge_packs", "$(hostname)-chunks-$(CHUNK_SIZE).jls"), output_chunks)
```

Comment:
I believe I gave some comments on this previously. Not sure where this would save, if RAGKit is this module, use `pkgdir(RAGKit)` or use relative path to the folder of the script file with `@__DIR__`